solr/contrib/ltr/src/test/org/apache/solr/ltr/TestLTROnSolrCloud.java

```diff
@@ -232,32 +230,47 @@ private void createCollection(String name, String config, int numShards, int num
     solrCluster.waitForActiveCollection(name, numShards, numShards * numReplicas);
   }
 
-
   void indexDocument(String collection, String id, String title, String description, int popularity)
     throws Exception{
     SolrInputDocument doc = new SolrInputDocument();
     doc.setField("id", id);
     doc.setField("title", title);
     doc.setField("description", description);
     doc.setField("popularity", popularity);
+    if(popularity != 1) {
+      // check that empty values will be read as default
+      doc.setField("dvIntField", popularity);
+      doc.setField("dvLongField", popularity);
+      doc.setField("dvFloatField", ((float) popularity) / 10);
+      doc.setField("dvDoubleField", ((double) popularity) / 10);
+      doc.setField("dvStrNumField", popularity);
+      doc.setField("dvStrBoolField", popularity % 2 == 0 ? "T" : "F");
+    }
     solrCluster.getSolrClient().add(collection, doc);
   }
 
   private void indexDocuments(final String collection)
        throws Exception {
     final int collectionSize = 8;
-    for (int docId = 1; docId <= collectionSize;  docId++) {
+    // put documents in reversed order to check that advanceExact is working correctly
+    for (int docId = collectionSize; docId >= 1;  docId--) {
```

Comment:
Similar to multi-segment case coverage below, randomisation could be used here to cover both ascending and descending order. Or another way could be to form a list of ids and to shuffle it with the randomiser and then to use it to index.

Reply:
I like your idea of shuffling the ids and used it in 4559415.